### PR TITLE
Implement initial Azure Media Service REST API communication

### PR DESCRIPTION
**Pull request by oksana-slu**
@kabirkhan @saravpal 
1. Extended native OpenEdx Organization model with `SettingsAzureOrganization` model;
2. Configured admin section for SettingsAzureOrganization model fields;
3. Created client to communicate with Azure Media Service API - `MediaServicesManagementClient`;
4. Implemented xBlock studio view to customize editing modal window behavior;
5. Switched from Mako to Django template engine;
6. Configured initial CircleCI service automatic checks;
7. New code base covered with unit tests.

### AMS content is shared per OpenEdx organization
Technically AMS API credentials are configured per Organization.
If there is no explicit configuration for the current organization - fallback to global (per OpenEdx installation) configuration is performed.
To configure global AMS API ones should add to `lms.env.json` into `FEATURES` section following settings:
```
"AZURE_CLIENT_ID": <azure-open-directory-app-id>
"AZURE_CLIENT_SECRET": <azure-open-directory-app-secret-key>
"AZURE_TENANT": <azure-open-directory-id>
"AZURE_REST_API_ENDPOINT": <azure-media-service-rest-api-endpoint>
```

### CircleCI service automatic checks include:
- python code style (flake8);
- javascript code style (eslint);
- python unit tests (nose, coverage);
- javascript unit tests (karma, jasmine).

To be able to run all checks manually ones should:

#### NOTE: assuming Ubuntu as a testing environment OS
- create and activate new python [`virtualenv`](https://virtualenv.pypa.io/en/stable/);
- change dir to project's root;
- install base system dependencies packages:
```
make install-tools
```
- install project dependencies:
```
make install-dev
```
- install testing dependencies:
```
make install-test
```
After that, it is possible to run project checks:
```
# run all together:
make all
# run unit tests only:
make test
# run code style (quality) checks only:
make quality
# help:
make help
```

### CODE REVIEW

**Comment by sdolenc** [comment]
I apologize it has taken me so long to review this. The PR was submitted while I was out on vacation so I missed it, but I should have time to review it this afternoon
**update**:
- I will not have time to finish reviewing this today, but I should be able to complete the review tomorrow. I'm impressed with the parts I've seen thus far and look forward to seeing the rest of it. Great work!
- I've also added @kabirkhan to the review in case he has the chance to take a look as well.

File: azure_media_services/static/js/spec/javascript-code-spec.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Tests for studio edit's utils
+ */
+/* global AzureMediaServicesBlock */
+
+describe('exspression: 1+1', function() {
```
Comment (sdolenc) [inline azure_media_services/static/js/spec/javascript-code-spec.js:6]: should be "expression"

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,4064 @@
+{
+  "requires": true,
```
Comment (sdolenc) [inline package-lock.json:2]: (not an issue) note to other code reviewers: you don't have to review this file. it is auto-generated. the accepted practice is to commit this to source control. future changes to this file will be much smaller

**Comment by kabirkhan** [comment]
Awesome initial work!

**Review by kabirkhan — APPROVED** [review]

**Review by sdolenc — APPROVED** [review]

**Pull request merged**